File: app/src/main/java/com/friendlypos/application/FriendlyApp.java
package com.friendlypos.application;

import android.app.Application;

import io.realm.Realm;
import io.realm.RealmConfiguration;

public class FriendlyApp extends Application {

    @Override
    public void onCreate() {
        super.onCreate();

        // Configure Realm for the application

        Realm.init(this);

        RealmConfiguration config = new RealmConfiguration.Builder()
                .name("app.realm")
                .schemaVersion(1)
                .deleteRealmIfMigrationNeeded()
                .build();
        Realm.setDefaultConfiguration(config);

    }
}

File: app/src/main/java/com/friendlypos/application/FriendlyApp.kt
package com.friendlysystemgroup.friendlypos.application

import android.app.Application
import com.friendlypos.util.AppContextProvider
import io.realm.Realm
import io.realm.RealmConfiguration

class FriendlyApp : Application() {
    override fun onCreate() {
        super.onCreate()

        // Inicializar AppContextProvider para reemplazar SyncObjectServerFacade
        AppContextProvider.init(this)
        
        Realm.init(this)

        val config = RealmConfiguration.Builder()
            .name("app.realm")
            .schemaVersion(1)
            .deleteRealmIfMigrationNeeded()
            .build()
        Realm.setDefaultConfiguration(config)
    }
}